src/base/bittorrent/torrentimpl.cpp

```diff
@@ -103,7 +104,14 @@ namespace
 
     QString toString(const lt::tcp::endpoint &ltTCPEndpoint)
     {
-        return QString::fromStdString((std::stringstream() << ltTCPEndpoint).str());
+        static QCache<lt::tcp::endpoint, QString> cache;
+
+        if (const QString *endpointName = cache.object(ltTCPEndpoint))
+            return *endpointName;
+
+        const QString endpointName = QString::fromStdString((std::ostringstream() << ltTCPEndpoint).str());
+        cache.insert(ltTCPEndpoint, new QString(endpointName));
```

Comment:
It would be even faster to use `QString::fromLatin1`:
```suggestion
        const std::string tmp = (std::ostringstream() << ltTCPEndpoint).str();
        const auto endpointName = QString::fromLatin1(tmp.c_str());
        cache.insert(ltTCPEndpoint, new QString(endpointName));
```

Reply:
> It would be even faster to use `QString::fromLatin1`:

Ok. Done.